src/main/java/com/hillel/items_exchange/security/jwt/DeletedUserFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.hillel.items_exchange.security.jwt;
+
+import com.hillel.items_exchange.model.User;
+import com.hillel.items_exchange.model.enums.Status;
+import com.hillel.items_exchange.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.filter.GenericFilterBean;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Optional;
+
+import static com.hillel.items_exchange.util.MessageSourceUtil.getMessageSource;
+import static com.hillel.items_exchange.util.MessageSourceUtil.getParametrizedMessageSource;
+
+@RequiredArgsConstructor
+public class DeletedUserFilter extends GenericFilterBean {
```

Comment:
```suggestion
@Component
@RequiredArgsConstructor
public class DeletedUserFilter extends GenericFilterBean {

    private final UserService userService;

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {

        HttpServletRequest servletRequest = (HttpServletRequest) request;
        Principal principal = servletRequest.getUserPrincipal();
        if (principal != null) {

            String username = principal.getName();
            Optional<User> selfDeletedUser = userService.findByUsernameOrEmail(username)
                    .filter(Predicate.not(User::isEnabled));
            if (selfDeletedUser.isPresent() && !isAllowedMethodsAccessed(servletRequest)) {
                blockFurtherAccessWithError(selfDeletedUser.get(), (HttpServletResponse) response);
                return;
            }
        }
        chain.doFilter(request, response);
    }

    private void blockFurtherAccessWithError(User user, HttpServletResponse response) throws IOException {
        try (PrintWriter writer = response.getWriter()) {
            writer.write(getMessageSource("exception.illegal.operation")
                    .concat(". ")
                    .concat(getParametrizedMessageSource("account.deleted.first", userService.getDaysBeforeDeletion(user))));
            response.setStatus(HttpServletResponse.SC_FORBIDDEN);
        }
    }

    private boolean isAllowedMethodsAccessed(HttpServletRequest request) {
        String requestedURL = request.getRequestURL().toString();
        String requestMethod = request.getMethod();
        return HttpMethod.PUT.matches(requestMethod) && requestedURL.contains("/user/service/restore") ||
                HttpMethod.GET.matches(requestMethod);
    }
}
```

Reply:
Additionally you needs to change User entity:

`public class User extends BaseEntity implements UserDetails {`

```
    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return Collections.singletonList(new SimpleGrantedAuthority(getRole().getName()));
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return this.getStatus() != Status.BANNED;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return this.getStatus() == Status.ACTIVE;
    }
```